_drafts/zero-to-main-2.md

```diff
@@ -0,0 +1,536 @@
+uninitialized static memory in. `.bss` must be reserved in the memory map, but there
+is nothing to load, as all variables are initialized to zero. As such, this is
+what it should look like:
+
+```
+SECTION {
+    ...
+    .bss (NOLOAD) :
+    {
+        *(.bss*)
+        *(COMMON)
+    } > ram
+}
+```
+
+You'll note that the `.bss` section also includess `*(COMMON)`. This is a
+special input section where the compiler puts global unitialized variables that
+go beyond file scope. `int foo;` goes there, while `static int foo;` does not.
+This allows the linker to merge multiple definitions into one symbol if they
+have the same name.
+
+We indicate that this section is not loaded with the `NOLOAD` property. This is
+the only section property used in modern linker scripts.
+
+We do the same thing for our `.stack` memory, since it is in RAM and not loaded.
+As the stack contains no symbols, we must explicitly reserve space for it by
+indicating its size. We also must align the stack on an 8-byte boundary per ARM
+Procedure Call Standards
+([AAPCS](https://static.docs.arm.com/ddi0403/ec/DDI0403E_c_armv7m_arm.pdf)).
+
+In order to achieve these goals, we turn to a special variable `.`, also known
+as the "location counter". The location counter tracks the current offset into a
+given memory region. As sections are added, the location counter increments
+accordingly. You can force alignemnt or gaps by setting the location counter
+forward. You may not set it backwards, and the linker will throw an error if you
+try.
+
+We set the location counter with the `ALIGN` function, to align the section, and
+use simple assignment and arithmetic to set the section size:
+
+```
+STACK_SIZE = 0x2000; /* 8 kB */
+
+SECTION {
+    ...
+    .stack (NOLOAD) :
+    {
+        . = ALIGN(8);
+        . = . + STACK_SIZE;
+        . = ALIGN(8);
+    } > ram
+    ...
+}
+```
+
+Only one to go!
+
+The `.data` section contains static variables which have an
+initial value at boot. You will remember from our previous article that since RAM
+isn't persisted while power is off, those sections need to be loaded in flash.
+At boot, the `Reset_Handler` copies the data from flash to RAM before the `main`
+function is called.
+
+To make this possible, every section in our linker script has two addresses,
+its *load* address (LMA) and its *virtual* address (VMA). In a firmware context,
+the LMA is where your JTAG loader needs to place the section and the VMA is
+where the section is found during execution.
+
+You can think of the LMA as the address "at rest" and the VMA the address during
+execution i.e. when the device is on and the program is running.
+
+The syntax to specify the LMA is relatively straightforward: every address is
+two part: <VMA> AT <LMA>. In our case it looks like this:
+
+```
+.data :
+{
+    *(.data*);
+} > ram AT > rom /* "> ram" is the VMA, "> rom" is the LMA */
+```
+
+Note that instead of appending a section to a memory region, you can explicity
```

Comment:
maybe: `you can explicity` -> `you could also explicitly`?  or just drop the block altogether since we don't use it anywhere in the final linker script

Reply:
changed!